bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/SncfBridgeHandler.java

```diff
@@ -0,0 +1,177 @@
+
+    public static final String SERVICE_URL = "https://api.sncf.com/v1/coverage/sncf/";
+
+    private final Logger logger = LoggerFactory.getLogger(SncfBridgeHandler.class);
+    private final LocationProvider locationProvider;
+    private final ExpiringCacheMap<String, String> cache = new ExpiringCacheMap<>(Duration.ofMinutes(1));
+    private final HttpClient httpClient;
+
+    private final Gson gson;
+    private @NonNullByDefault({}) String apiId;
+
+    public SncfBridgeHandler(Bridge bridge, Gson gson, LocationProvider locationProvider, HttpClient httpClient) {
+        super(bridge);
+        this.locationProvider = locationProvider;
+        this.httpClient = httpClient;
+        this.gson = gson;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SNCF API bridge handler.");
+        apiId = (String) getConfig().get("apiID");
+        if (apiId != null && apiId.length() != 0) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/null-or-empty-api-key");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("SNCF API Bridge is read-only and does not handle commands");
+    }
+
+    private <T extends SncfAnswer> T getResponseFromCache(String url, Class<T> objectClass) throws SncfException {
+        String answer = cache.get(url);
+        if (answer == null) {
+            final String newAnswer = getResponse(url);
+            cache.put(url, () -> newAnswer);
+            answer = newAnswer;
+        }
+        @Nullable
+        T response = gson.fromJson(answer, objectClass);
+        if (response == null) {
+            throw new SncfException("Unable to deserialize API answer");
+        }
+        if (response.message != null) {
+            throw new SncfException(response.message);
+        }
+        return response;
+    }
+
+    private String getResponse(String url) throws SncfException {
+        try {
+            logger.debug("SNCF Api request: URL = '{}'", url);
+            ContentResponse contentResponse = httpClient.newRequest(url).method(GET).timeout(10, TimeUnit.SECONDS)
+                    .header("Authorization", apiId).send();
+            int httpStatus = contentResponse.getStatus();
+            String content = contentResponse.getContentAsString();
+            logger.debug("SNCF Api response: status = {}, content = '{}'", httpStatus, content);
+            switch (httpStatus) {
+                case OK_200:
+                    return content;
+                case BAD_REQUEST_400:
+                case UNAUTHORIZED_401:
+                case NOT_FOUND_404:
+                    logger.debug("SNCF Api server responded with status code {}: {}", httpStatus, content);
+                    throw new SncfException(content);
```

Comment:
As the code is the same as the default, the code for these 3 cases can be removed.

Reply:
Done